src/app/dataValidator/helpers/dataLoader/readdirSync.test.js

```diff
@@ -18,24 +21,25 @@ const expectMethodToBeCalledTimes = (
 
 describe('readdirSync helper', () => {
   it('should call readScenario for every file in the /data directory', () => {
-    const readScenarioSpy = jest.spyOn(readScenario, 'readScenario');
+    readScenarioSpy = jest.spyOn(readScenario, 'readScenario');
 
-    expectMethodToBeCalledTimes(41, readScenarioSpy);
+    expectMethodToBeCalledTimes(79, readScenarioSpy);
   });
 
   it('should call fileToValidate for only the valid json file in the /data directory', () => {
-    const fileToValidateSpy = jest.spyOn(readScenario, 'fileToValidate');
+    fileToValidateSpy = jest.spyOn(readScenario, 'fileToValidate');
 
-    expectMethodToBeCalledTimes(25, fileToValidateSpy);
+    expectMethodToBeCalledTimes(60, fileToValidateSpy);
   });
 
   it('should call fileToValidate for only the files in /data/prod/news', () => {
-    const fileToValidateSpy = jest.spyOn(readScenario, 'fileToValidate');
+    fileToValidateSpy.mockRestore();
```

Comment:
This is removing the mock, but on the next line you are setting the mock up in the same way as you do above. Is that intended?

Reply:
Correct, the usual pattern is to put mockRestores either after an expect or between tests using `afterEach` however due to the async nature of the test restoring affect the expect actually breaks the test and using `afterEach` is only valid if all the tests mock the same methods. 

Other solutions could be: 
- to globally scope the variables and use a consistent `afterEach` mock restore approach
- delegate the mocking to a function which first attempts a mock restore before setting the new mock

Reply:
Why do the mocks need to be restored between test executions in the first place? Could the spy just be set in a scope that includes both these tests?

Reply:
Without the mock restore the spy doesn't reset (I'm unsure as to why because it should only be considered in the functional scope), this can be seen as the test which checks that `data/prod/news` only has 2 scenarios errors thinking the method was called 62 times. The additional 60 is coming from the previous test, see screenshot:
![image](https://user-images.githubusercontent.com/7791726/45813045-74461b00-bcc9-11e8-8e68-da3d48650a43.png)